esphome/components/ft63x6/ft63x6.cpp

```diff
@@ -0,0 +1,165 @@
+
+void FT63X6Touchscreen::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up FT63X6Touchscreen Touchscreen...");
+  if (this->interrupt_pin_ != nullptr) {
+    this->interrupt_pin_->pin_mode(gpio::FLAG_INPUT | gpio::FLAG_PULLUP);
+    this->interrupt_pin_->setup();
+    this->store_.pin = this->interrupt_pin_->to_isr();
+    this->interrupt_pin_->attach_interrupt(FT63X6TouchscreenStore::gpio_intr, &this->store_,
+                                           gpio::INTERRUPT_FALLING_EDGE);
+  }
+
+  if (this->reset_pin_ != nullptr) {
+    this->reset_pin_->setup();
+  }
+
+  this->hard_reset_();
+
+  // Get touch resolution
+  this->x_resolution_ = 320;
+  this->y_resolution_ = 480;
+}
+
+void FT63X6Touchscreen::loop() {
+  if (this->interrupt_pin_ == nullptr || !this->store_.touch)
+    return;
+  this->store_.touch = false;
+  check_touch_();
+}
+
+void FT63X6Touchscreen::update() {
+  if (this->interrupt_pin_ == nullptr)
+    check_touch_();
+}
+
+void FT63X6Touchscreen::check_touch_() {
+  touch_point_.touch_count = read_touch_count_();
+
+  uint8_t touch_id = 0;
+  if (touch_point_.touch_count >= 1) {
+    touch_id = read_touch_id_(FT63X6_ADDR_TOUCH1_ID);  // id1 = 0 or 1
+    touch_point_.tp[touch_id].status = (touch_point_.tp[touch_id].status == TouchStatusEnum::RELEASE)
+                                           ? TouchStatusEnum::TOUCH
+                                           : TouchStatusEnum::REPEAT;
+    touch_point_.tp[touch_id].x = read_touch_coordinate_(FT63X6_ADDR_TOUCH1_X);
+    touch_point_.tp[touch_id].y = read_touch_coordinate_(FT63X6_ADDR_TOUCH1_Y);
+  } else {
+    touch_point_.tp[touch_id].status = TouchStatusEnum::RELEASE;
+  }
+
+  if (touch_point_.touch_count >= 2) {
+    touch_id = read_touch_id_(FT63X6_ADDR_TOUCH2_ID);  // id2 = 0 or 1(~id1 & 0x01)
+    touch_point_.tp[touch_id].status = (touch_point_.tp[touch_id].status == TouchStatusEnum::RELEASE)
+                                           ? TouchStatusEnum::TOUCH
+                                           : TouchStatusEnum::REPEAT;
+    touch_point_.tp[touch_id].x = read_touch_coordinate_(FT63X6_ADDR_TOUCH2_X);
+    touch_point_.tp[touch_id].y = read_touch_coordinate_(FT63X6_ADDR_TOUCH2_Y);
+  } else {
+    touch_point_.tp[~touch_id & 0x01].status = TouchStatusEnum::RELEASE;
+  }
+
+  if (touch_point_.touch_count == 0) {
+    for (auto *listener : this->touch_listeners_)
+      listener->release();
+    return;
+  }
+
+  std::vector<TouchPoint> touches;
+  uint8_t touch_count = std::min<uint8_t>(touch_point_.touch_count, 2);
+  ESP_LOGV(TAG, "Touch count: %d", touch_count);
+
+  uint16_t w = this->display_->get_width_internal();
+  uint16_t h = this->display_->get_height_internal();
+
+  for (int i = 0; i < touch_count; i++) {
+    uint32_t raw_x = touch_point_.tp[i].x * w / this->x_resolution_;
+    uint32_t raw_y = touch_point_.tp[i].y * h / this->y_resolution_;
+
+    TouchPoint tp;
+    switch (this->display_->get_rotation()) {
```

Comment:
i think it needs to be the code below: as it is in the XPT2046 touchscreen driver as well
```suggestion
   switch (static_cast<TouchRotation>(this->display_->get_rotation())) {
```

Reply:
This and the codeowner issue were fixed at a later commit

Reply:
Actually, this got back with another lint issue. Will fix